pkg/cloudprovider/provider/linode/provider.go

```diff
@@ -0,0 +1,465 @@
+package linode
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/linode/linodego"
+	"golang.org/x/oauth2"
+
+	"github.com/kubermatic/machine-controller/pkg/cloudprovider/cloud"
+	"github.com/kubermatic/machine-controller/pkg/cloudprovider/common/ssh"
+	cloudprovidererrors "github.com/kubermatic/machine-controller/pkg/cloudprovider/errors"
+	"github.com/kubermatic/machine-controller/pkg/cloudprovider/instance"
+	"github.com/kubermatic/machine-controller/pkg/providerconfig"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	common "sigs.k8s.io/cluster-api/pkg/apis/cluster/common"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+type provider struct {
+	configVarResolver *providerconfig.ConfigVarResolver
+}
+
+// New returns a linode provider
+func New(configVarResolver *providerconfig.ConfigVarResolver) cloud.Provider {
+	return &provider{configVarResolver: configVarResolver}
+}
+
+type RawConfig struct {
+	Token             providerconfig.ConfigVarString   `json:"token"`
+	Region            providerconfig.ConfigVarString   `json:"region"`
+	Type              providerconfig.ConfigVarString   `json:"type"`
+	Backups           providerconfig.ConfigVarBool     `json:"backups"`
+	IPv6              providerconfig.ConfigVarBool     `json:"ipv6"`
+	PrivateNetworking providerconfig.ConfigVarBool     `json:"private_networking"`
+	Tags              []providerconfig.ConfigVarString `json:"tags"`
+}
+
+type Config struct {
+	Token             string
+	Region            string
+	Type              string
+	Backups           bool
+	IPv6              bool
+	PrivateNetworking bool
+	Monitoring        bool
```

Comment:
The `Monitoring` option isn't used anywhere or did I miss it?

Reply:
Nice catch. Linode's has a few options for this, alerting on CPU,net,disk usage threaholds, auto rebooting on shutdown/panic (Lassie), and Longview (graphs and stats, process/package/service monitoring).  

For the sake of this PR, that field should be removed.